bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/connection/HubCommunicationThread.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nobohub.internal.connection;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.nobohub.internal.NoboHubBindingConstants;
+import org.openhab.binding.nobohub.internal.NoboHubBridgeHandler;
+import org.openhab.binding.nobohub.internal.model.NoboCommunicationException;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Thread that reads from the Nobø Hub and sends HANDSHAKEs to keep the connection open.
+ *
+ * @author Jørgen Austvik - Initial contribution
+ */
+@NonNullByDefault
+public class HubCommunicationThread extends Thread {
+
+    private enum HubCommunicationThreadState {
+        STARTING(ThingStatus.ONLINE, ThingStatusDetail.UNKNOWN.NONE, "@text/message.bridge.status.connecting"),
+        CONNECTED(ThingStatus.ONLINE, ThingStatusDetail.NONE, ""),
+        DISCONNECTED(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/message.bridge.status.failed"),
+        STOPPED(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "");
```

Comment:
Sorry, still not correct. **BRIDGE_OFFLINE** is used for a thing when its bridge is offline, which then will be set by the framework. For the bridge itself, consider **COMMUNICATION_ERROR**. Now I tracked it and see that `STOPPED` is used only from bridge handler's `dispose()` method, so actually you don't need to set status as the framework will automatically do this.

Reply:
Refactoring is not always easy. How about now?

Reply:
Resolving comment after your fix, the remaining potential problem is covered by other comment.